cloudevents/http/dict_methods.py

```diff
@@ -0,0 +1,44 @@
+#  Copyright 2018-Present The CloudEvents Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import datetime
+import enum
+import typing
+
+from cloudevents.http.event import CloudEvent
+
+
+def encode_attribute_value(value: typing.Any) -> typing.Any:
+    if isinstance(value, enum.Enum):
+        return value.value
+    if isinstance(value, datetime.datetime):
+        return value.isoformat()
+
+    return value
+
+
+def from_dict(event: typing.Dict[str, typing.Any]) -> CloudEvent:
+    return CloudEvent(
+        {
+            attr_name: encode_attribute_value(attr_value)
+            for attr_name, attr_value in event.items()
+            if attr_name != "data"
+        },
+        data=event.get("data"),
+    )
+
+
+def to_dict(event: CloudEvent) -> typing.Dict[str, typing.Any]:
+    result = {attribute_name: event.get(attribute_name) for attribute_name in event}
+    result["data"] = event.data
+    return result
```

Comment:
Same here. I guess it may be much easier to use if such methods are class/instance methods of the CloudEvent instance itself.
Or we may just have both a helper function alias and an instance method.